v2/datastream-common/src/main/java/com/google/cloud/teleport/v2/datastream/sources/DataStreamIO.java

```diff
@@ -172,6 +174,23 @@ public DataStreamIO withDirectoryWatchDuration(Duration directoryWatchDuration)
     return this;
   }
 
+  /**
+   * Set applyReshuffle to false in order to skip the reshuffle step on Datastream records.
+   *
+   * <p>In the Datastream to Spanner template, one reshuffle step is already performed to balance
+   * the load across workers. Load testing showed that removing this second reshuffle did not impact
+   * performance but significantly reduced pipeline costs.
+   *
+   * <p>Since the cost of a pipeline depends on the amount of data processed, reshuffling increases
+   * data processing due to extra grouping and shuffling across workers. By skipping this extra
+   * reshuffle, we reduce the total data processed, leading to cost savings while maintaining
+   * performance.
```

Comment:
Makes sense, but you should also talk about trade-off between distributing records across workers and reshuffling costs and overhead. Maybe link the documentation to reshuffling here.